### PR TITLE
Fourth task

### PR DESCRIPTION
**Pull request by an2onsol**
https://projecteuler.net/problem=4

### CODE REVIEW

File: fourth_problem.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+          max_palindrome = m
+      end
+      b -= 1
+    end
+    a -= 1
+  end
+  return max_palindrome
+end
+
+puts largest_palindrome
```
Comment (AndreyAnischik) [inline fourth_problem.rb:19]: add line

File: fourth_problem_spec.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'rspec'
+require_relative './fourth_problem'
+
+describe '#largest_palindrome' do
+  it 'should return the largest palindrome' do
+    expect(largest_palindrome).to eq 906609
+  end
+end
```
Comment (AndreyAnischik) [inline fourth_problem_spec.rb:8]: add line

File: fourth_problem.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+def largest_palindrome
+  a = 999
```
Comment (AndreyAnischik) [inline fourth_problem.rb:2]: move it to constant and give it a reasonable name

File: fourth_problem.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  max_palindrome = 0
```
Comment (AndreyAnischik) [inline fourth_problem.rb:3]: name it like constant, https://github.com/airbnb/ruby#naming

File: fourth_problem.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    while b > 100
+      m = a * b
+      if m.to_s.reverse == m.to_s && m > max_palindrome
+          max_palindrome = m
+      end
+      b -= 1
+    end
+    a -= 1
+  end
+  return max_palindrome
```
Comment (AndreyAnischik) [inline fourth_problem.rb:16]: no need to use return

File: fourth_problem.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+def largest_palindrome
+  a = 999
+  max_palindrome = 0
+
+  while a > 100
+    b = a
+    while b > 100
+      m = a * b
+      if m.to_s.reverse == m.to_s && m > max_palindrome
+          max_palindrome = m
```
Comment (AndreyAnischik) [inline fourth_problem.rb:10]: too many tabs here

File: fourth_problem.rb [hunk 7]
```diff
@@ -0,0 +1,21 @@
+MAX_THREE_DIGIT = 999
+
+def largest_palindrome
+  first_factor = MAX_THREE_DIGIT
```
Comment (AndreyAnischik) [inline fourth_problem.rb:4]: no need to assign it here, just use it where you need

**Review by AndreyAnischik — APPROVED** [review]

**Review by bronium — APPROVED** [review]

**Pull request closed without merging**